imap_processing/tests/spice/test_pointing_frame.py

```diff
@@ -0,0 +1,185 @@
+)
+
+
+@pytest.fixture
+def pointing_frame_kernels(spice_test_data_path):
+    """List SPICE kernels."""
+    required_kernels = [
+        "imap_science_0001.tf",
+        "imap_sclk_0000.tsc",
+        "imap_sim_ck_2hr_2secsampling_with_nutation.bc",
+        "imap_wkcp.tf",
+        "naif0012.tls",
+    ]
+    kernels = [str(spice_test_data_path / kernel) for kernel in required_kernels]
+    return kernels
+
+
+@pytest.fixture
+def et_times(pointing_frame_kernels):
+    """Tests get_et_times function."""
+    spiceypy.furnsh(pointing_frame_kernels)
+
+    ck_kernel, _, _, _ = spiceypy.kdata(0, "ck")
+    ck_cover = spiceypy.ckcov(ck_kernel, -43000, True, "INTERVAL", 0, "TDB")
+    et_start, et_end = spiceypy.wnfetd(ck_cover, 0)
+
+    # 1 spin/15 seconds; 10 quaternions / spin.
+    num_samples = (et_end - et_start) / 15 * 10
+    # There were rounding errors when using spiceypy.pxform so np.ceil and np.floor
+    # were used to ensure the start and end times were within the ck range.
+    et_times = np.linspace(
+        np.ceil(et_start * 1e6) / 1e6, np.floor(et_end * 1e6) / 1e6, int(num_samples)
+    )
+
+    return et_times
+
+
+@pytest.fixture
+def fake_repoint_data(monkeypatch, spice_test_data_path):
+    """Generate fake spin dataframe for testing"""
+    fake_repoint_path = spice_test_data_path / "fake_repoint_data.csv"
+    monkeypatch.setenv("REPOINT_DATA_FILEPATH", str(fake_repoint_path))
+    return fake_repoint_path
+
+
+def test_average_quaternions(et_times, pointing_frame_kernels):
+    """Tests average_quaternions function."""
+    spiceypy.furnsh(pointing_frame_kernels)
+    q_avg = _average_quaternions(et_times)
+
+    # Generated from MATLAB code results
+    q_avg_expected = np.array([-0.6611, 0.4981, -0.5019, -0.2509])
+    np.testing.assert_allclose(q_avg, q_avg_expected, atol=1e-4)
+
+
+def test_create_rotation_matrix(et_times, pointing_frame_kernels):
+    """Tests create_rotation_matrix function."""
+    spiceypy.furnsh(pointing_frame_kernels)
+    rotation_matrix = _create_rotation_matrix(et_times)
+    q_avg = _average_quaternions(et_times)
+    z_avg = spiceypy.q2m(list(q_avg))[:, 2]
+
+    rotation_matrix_expected = np.array(
+        [[0.0000, 0.0000, 1.0000], [0.9104, -0.4136, 0.0000], [0.4136, 0.9104, 0.0000]]
+    )
+    z_avg_expected = np.array([0.4136, 0.9104, 0.0000])
+
+    np.testing.assert_allclose(z_avg, z_avg_expected, atol=1e-4)
+    np.testing.assert_allclose(rotation_matrix, rotation_matrix_expected, atol=1e-4)
+
+
+def test_create_pointing_frame(
+    spice_test_data_path, pointing_frame_kernels, tmp_path, et_times, fake_repoint_data
+):
+    """Tests create_pointing_frame function."""
+
+    # This is how the repoint data is generated.
+    # We will use fake data for now to match the coverage of the attitude kernel.
+    # repoint_df = get_repoint_data()
+    # repoint_start_met = repoint_df["repoint_start_met"].values
+    # repoint_end_met = repoint_df["repoint_end_met"].values
```

Comment:
Just a note that I don't think this will be quite right, I think it will actually need to be inverted. Our repointing file gives us the start and stop of the repointing maneuver (so ~15 minutes period). I think we'll want something like going from the end of the first repointing to the start of the second repointing. But we'll also need to handle some other potential edge cases like if a repointing maneuver hasn't finished.
```suggestion
    # repoint_df = get_repoint_data()
    # repoint_start = repoint_df["repoint_end_met"].values[:-1]
    # repoint_end_met = repoint_df["repoint_start_met"].values[1:]
```

Reply:
I wonder if we should have a method on the table that returns an iterable of the valid repointings so this is more re-usable. (not here)